Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -140,6 +148,26 @@ private void OnPostFlash(EntityUid uid, HeadRevolutionaryComponent comp, ref Aft
             return;
         }
 
+        // Check if the user has a ban on "Revolutionary"
+        if (mind != null && mind.Session != null)
+        {
+            var antagBans = _banManager.GetAntagBans(mind.Session.UserId);
+            var antagAllSelection = Loc.GetString("ban-panel-role-selection-antag-all-option");
+            var revolutionaryRoleId = "Rev"; // Assuming "Rev" is the ID for RevolutionaryRoleComponent
+
+            if (antagBans != null && (antagBans.Contains(antagAllSelection) || antagBans.Contains(revolutionaryRoleId)))
+            {
+                // Notify the player and trigger a random bad event
+                _popup.PopupEntity("You have been converted but are unable to play due to a ban for this role.", ev.Target, ev.Target, PopupType.LargeCaution);
+
+                var randomDelay = new Random().Next(10000, 60000); // 10-60 seconds
+                var @event = ev;
+                Timer.Spawn(TimeSpan.FromMilliseconds(randomDelay), () => _adminVerbSystem.RandomDeath(@event.Target));
+
```

Comment:
Please do not use timers for gamesim stuff.

Reply:
Moved it in to ActiveTick updates now.